import-marc.sh

```diff
@@ -153,16 +155,31 @@ then
   mkdir -p $SOLRJ_DIR
   for file in $VUFIND_HOME/solr/vendor/server/solr-webapp/webapp/WEB-INF/lib/solr*.jar $VUFIND_HOME/solr/vendor/server/solr-webapp/webapp/WEB-INF/lib/http*.jar
   do
-    ln -s $file $SOLRJ_DIR/`basename $file`
+    ln -s $file $SOLRJ_DIR/`basename "$file"`
   done
 fi
 
 #####################################################
 # Execute Importer
 #####################################################
 
-RUN_CMD="$JAVA $INDEX_OPTIONS -Duser.timezone=UTC -Dlog4j.configuration=file://$LOG4J_CONFIG $EXTRA_SOLRMARC_SETTINGS -jar $JAR_FILE $PROPERTIES_FILE -solrj $SOLRJ_DIR -lib_local "$VUFIND_HOME/import/lib_local\;$VUFIND_HOME/solr/vendor/modules/analysis-extras/lib" $NORMALIZED_PATHS"
-echo "Now Importing $NORMALIZED_PATHS ..."
-# solrmarc writes log messages to stderr, write RUN_CMD to the same place
-echo "`date '+%h %d, %H:%M:%S'` $RUN_CMD" >&2
-exec $RUN_CMD
+# Build the command as an array
+RUN_CMD=(
+  "$JAVA"
+  $INDEX_OPTIONS
+  -Duser.timezone=UTC
+  -Dlog4j.configuration="file://$LOG4J_CONFIG"
+  $EXTRA_SOLRMARC_SETTINGS
+  -jar "$JAR_FILE"
+  "$PROPERTIES_FILE"
+  -solrj "$SOLRJ_DIR"
+  -lib_local "$VUFIND_HOME/import/lib_local;$VUFIND_HOME/solr/vendor/modules/analysis-extras/lib"
+  "${NORMALIZED_PATHS[@]}"
+)
+
+# Debugging output
+echo "Now Importing: ${NORMALIZED_PATHS[*]}"
+echo "$(date '+%h %d, %H:%M:%S') ${RUN_CMD[*]}" >&2
+
+# Execute the command using the array
+exec "${RUN_CMD[@]}"
```

Comment:
Might be good to add an end-of-file carriage return here for stylistic consistency.

Reply:
Yup, good point...done.